src/Sema.zig

```diff
@@ -21771,10 +21771,10 @@ fn zirErrorCast(sema: *Sema, block: *Block, extended: Zir.Inst.Extended.InstData
     // operand must be defined since it can be an invalid error value
     const maybe_operand_val = try sema.resolveDefinedValue(block, operand_src, operand);
 
-    if (disjoint: {
+    const disjoint = disjoint: {
```

Comment:
The logic in this block should be skipped for error unions.

Reply:
I use it to avoid an `error_set_has_value` in the safety check but it could be removed too.

Reply:
Ah, my mistake